lua/nvim-tree/watcher.lua

```diff
@@ -0,0 +1,82 @@
+local uv = vim.loop
+
+local log = require "nvim-tree.log"
+local utils = require "nvim-tree.utils"
+
+local M = {}
+local Watcher = {}
+Watcher.__index = Watcher
+
+-- TODO check for a Watcher with the same opts.absolute_path and return that
+function Watcher.new(opts)
+  if not M.enabled then
+    return nil
+  end
+  log.line("watcher", "Watcher:new   '%s'", opts.absolute_path)
+
+  local stat, _ = uv.fs_stat(opts.absolute_path)
+  if not stat or stat.type ~= "directory" then
+    return nil
+  end
```

Comment:
not sure about this check, it is already done in the caller i believe

Reply:
done